### PR TITLE
Upgrade Go to v1.17

### PR DESCRIPTION
**Pull request by cimnine**
## Summary

Updates:

- Go to 1.17
- Kustomize to v4
- setup-envtest to latest

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`,
      as they show up in the changelog
- [ ] Update the documentation.
- [ ] Update tests.
- [ ] Link this PR to related issues.

### CODE REVIEW

**Review by ccremer — COMMENTED** [review]
@Kidswiss WDYT?
Does that make sense or is it too early?

File: operator/executor/cleaner/cleaner_integration_test.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
+//go:build integration
 // +build integration
```
[the remark this answers is not in the document]
Comment (cimnine) [reply operator/executor/cleaner/cleaner_integration_test.go:2]: See https://tip.golang.org/doc/go1.17#go-command

> The go command now understands //go:build lines and prefers them over // +build lines. The new syntax uses boolean expressions, just like Go, and should be less error-prone. As of this release, the new syntax is fully supported, and all Go files should be updated to have both forms with the same meaning. To aid in migration, gofmt now automatically synchronizes the two forms. For more details on the syntax and migration plan, see https://golang.org/design/draft-gobuild.

File: go.mod [hunk 2]
```diff
@@ -16,14 +16,114 @@ require (
...
+	github.com/BurntSushi/toml v0.3.1 // indirect
+	github.com/Masterminds/goutils v1.1.0 // indirect
+	github.com/Masterminds/semver v1.5.0 // indirect
+	github.com/Masterminds/sprig v2.22.0+incompatible // indirect
+	github.com/PuerkitoBio/purell v1.1.1 // indirect
+	github.com/PuerkitoBio/urlesc v0.0.0-20170810143723-de5bf2ad4578 // indirect
+	github.com/alessio/shellescape v1.4.1 // indirect
+	github.com/asaskevich/govalidator v0.0.0-20190424111038-f61b66f89f4a // indirect
+	github.com/beorn7/perks v1.0.1 // indirect
+	github.com/cespare/xxhash/v2 v2.1.1 // indirect
```
[the remark this answers is not in the document]
Comment (cimnine) [reply go.mod:42]: Not sure, it's what `go mod tidy` spits out.

File: operator/executor/cleaner/cleaner_integration_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ccremer) [reply operator/executor/cleaner/cleaner_integration_test.go:2]: Does that mean we need both or is it just for migration?
E.g. if every Dev would switch to Go 1.17, still needed?

[the remark this answers is not in the document]
Comment (cimnine) [reply operator/executor/cleaner/cleaner_integration_test.go:2]: > … and all Go files should be updated to have both forms with the same meaning.

I read this that for the time both are expected.

[the remark this answers is not in the document]
Comment (cimnine) [reply operator/executor/cleaner/cleaner_integration_test.go:2]: Yes, for now this is the expected behaviour, as described in https://golang.org/design/draft-gobuild#transition.

[the remark this answers is not in the document]
Comment (ccremer) [reply operator/executor/cleaner/cleaner_integration_test.go:2]: ah, thanks for summarizing

File: go.mod (hunk 2 above)
[the remark this answers is not in the document]
Comment (ccremer) [reply go.mod:42]: I don't like this, but I guess if we remove them the next Renovate PR to update dependencies would add it again...
I hope sometime it goes away, it looks just like bloat.

**Review by ccremer — APPROVED** [review]

File: go.mod (hunk 2 above)
[the remark this answers is not in the document]
Comment (cimnine) [reply go.mod:42]: We'll get rid of some of those when we update the Kubernetes modules to >v0.22.

**Pull request merged**